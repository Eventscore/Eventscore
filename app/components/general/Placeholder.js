//<Placeholder text={"Hello World"} />

import React, { PropTypes } from 'react';
import { View } from 'react-native';

// Consts and Libs
import { AppStyles } from '@theme/';

// Components
import { Text } from '@ui/';

const Placeholder = ({ text }) => (
  <View style={[AppStyles.container, AppStyles.containerCentered]}>
    <Text>{text}</Text>
  </View>
);

Placeholder.propTypes = { text: PropTypes.string };
Placeholder.defaultProps = { text: 'Coming soon...' };
Placeholder.componentName = 'Placeholder';

export default Placeholder;
