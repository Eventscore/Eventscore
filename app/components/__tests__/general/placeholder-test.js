/**
 * Test to check if the component renders correctly
 */
/* global it expect */
import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';

import Placeholder from '@components/general/Placeholder';

it('Placeholder renders correctly', () => {
  const tree = renderer.create(
    <Placeholder />,
  ).toJSON();

  expect(tree).toMatchSnapshot();
});

it('Placeholder w/ Text renders correctly', () => {
  const tree = renderer.create(
    <Placeholder text={'Hello world!'} />,
  ).toJSON();

  expect(tree).toMatchSnapshot();
});
