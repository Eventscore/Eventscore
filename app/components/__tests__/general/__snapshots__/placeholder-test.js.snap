// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Placeholder renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#E9EBEE",
        "flex": 1,
        "flexDirection": "column",
        "position": "relative",
      },
      Object {
        "alignItems": "center",
        "justifyContent": "center",
      },
    ]
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    h1={false}
    h2={false}
    h3={false}
    h4={false}
    h5={false}
    onPress={null}
    p={false}
    style={
      Array [
        Object {
          "color": "#222222",
          "fontFamily": "HelveticaNeue",
          "fontSize": 14,
          "fontWeight": "500",
          "lineHeight": 18,
        },
      ]
    }
  >
    Coming soon...
  </Text>
</View>
`;

exports[`Placeholder w/ Text renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#E9EBEE",
        "flex": 1,
        "flexDirection": "column",
        "position": "relative",
      },
      Object {
        "alignItems": "center",
        "justifyContent": "center",
      },
    ]
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    h1={false}
    h2={false}
    h3={false}
    h4={false}
    h5={false}
    onPress={null}
    p={false}
    style={
      Array [
        Object {
          "color": "#222222",
          "fontFamily": "HelveticaNeue",
          "fontSize": 14,
          "fontWeight": "500",
          "lineHeight": 18,
        },
      ]
    }
  >
    Hello world!
  </Text>
</View>
`;
